pkg/workspace/plugins.go

```diff
@@ -133,12 +133,23 @@ func (info *PluginInfo) SetFileMetadata(path string) error {
 // Install installs a plugin's tarball into the cache.  It validates that plugin names are in the expected format.
 func (info PluginInfo) Install(tarball io.ReadCloser) error {
 	// Fetch the directory into which we will expand this tarball, and create it.
-	pluginDir, err := info.DirPath()
+	finalDir, err := info.DirPath()
 	if err != nil {
 		return err
 	}
-	if err = os.MkdirAll(pluginDir, 0700); err != nil {
-		return errors.Wrapf(err, "creating plugin directory %s", pluginDir)
+
+	// We unpack the tarball into a temporary directory in the plugin cache and after that has completed rename it
+	// to prevent posioning the cache if the install is CTRL-C'd,
+	tempDir := finalDir + ".tmp"
```

Comment:
Good question.  It would not because of the `.tmp` suffix.  The regular expression we use for validating a plugin folder requires it not having this (see https://github.com/pulumi/pulumi/blob/abb0c2769baabf6648522ba235d4df73ac32ecaf/pkg/workspace/plugins.go#L353-L356)